app/services/arcgis_api/geocoder.rb

```diff
@@ -0,0 +1,58 @@
+module ArcgisApi
+  class Geocoder
+    def suggest(text)
+      url = "#{root_url}/suggest"
+      params = {
+        text: text,
+        category: 'address',
+        countryCode: 'USA',
+        f: 'json',
+      }
+
+      parse_suggestions(
+        faraday.get(url, params) do |req|
+          req.options.context = { service_name: 'arcgis_geocoder_suggest' }
+        end.body,
+      )
+    end
+
+    def findAddressCandidates
+    end
+
+    private
+
+    def root_url
+      IdentityConfig.store.arcgis_api_root_url
+    end
+
+    def faraday
+      Faraday.new(headers: request_headers) do |conn|
+        # Log request metrics
+        conn.request :instrumentation, name: 'request_metric.faraday'
+
+        # Raise an error subclassing Faraday::Error on 4xx, 5xx, and malformed responses
+        # Note: The order of this matters for parsing the error response body.
+        conn.response :raise_error
+
+        # Convert body to JSON
+        conn.request :json
+
+        # Parse JSON responses
+        conn.response :json
+      end
+    end
+
+    def request_headers
+      { 'Authorization' => "Bearer #{IdentityConfig.store.arcgis_api_key}" }
+    end
```

Comment:
Do we know if the API keys are long-lived? I was wondering if we'd instead need to add the username / password credentials as configuration and then make a request to create the API key as part of the code here, similar to what we do in the USPS proofer to retrieve a token.